### DISCUSSION TITLE
RTL password change

### DISCUSSION THREAD

**Discussion by jr46812000** [Ride-The-Lightning/RTL#1194] [Troubleshooting Questions] [answered]
Hello,
Can anyone tell me how to change the password on Ride The lightning. I would like to use it in umbrel, but the password is a long string of charachters. I need to change it so I am able to fund my lightning channel. On [I Have No Idea What I'm Doing: Lightning on Umbrel⚡️](https://stonly.com/guide/en/i-have-no-idea-what-im-doing-lightning-on-umbrel-enifmKFdle/Steps/269242), they said the password was moneyprintergobrr and I have seen this on these boards also, but mine was a long string that i copied and pasted.

**Comment by saubyk** [comment, marked as answer]
Hi @jr46812000 on Umbrel, you need to always use the password set in the app store. The application level password control is disabled in Umbrel, per their own security policy

**Reply by ghost** [reply]
What about running RTL on bare metal? How to change its passwd? just found it: https://github.com/Ride-The-Lightning/RTL/discussions/1400#discussioncomment-9585524

**Reply by ghost** [reply]
I am helping a friend setup a node and he has the opposite problem, after setting the passwd in the "multiPass" line it does not become "multiPasshashed", what could it be?

**Reply by saubyk** [reply]
has the node been restarted, after making the change?

**Reply by saubyk** [reply]
sorry, I mean RTL, not the node

**Reply by ghost** [reply]
Yes, even restarted the node to make sure it was not some related service that needed to be restarted. It's on a ubuntu server 24.04 and nodejs 20, not sure if all compatible.

**Reply by saubyk** [reply]
are you getting any error in the RTL log? If RTL has been restarted successfully, it should hash and store the hash of the password

**Reply by ghost** [reply]
I will post some logs in the next hours, thanks for the help so far. But no, it does not start with "node rtl"

**Reply by ghost** [reply]
$ sudo journalctl -f -u rtl
Jun 19 21:02:12 droidu node[25691]:     at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:383:12)
Jun 19 21:02:12 droidu node[25691]:     at ModuleLoader.resolve (node:internal/modules/esm/loader:352:25)
Jun 19 21:02:12 droidu node[25691]:     at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:227:38)
Jun 19 21:02:12 droidu node[25691]:     at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:87:39)
Jun 19 21:02:12 droidu node[25691]:     at link (node:internal/modules/esm/module_job:86:36) {
Jun 19 21:02:12 droidu node[25691]:   code: 'ERR_MODULE_NOT_FOUND'
Jun 19 21:02:12 droidu node[25691]: }
Jun 19 21:02:12 droidu node[25691]: Node.js v20.14.0
Jun 19 21:02:12 droidu systemd[1]: rtl.service: Main process exited, code=exited, status=1/FAILURE
Jun 19 21:02:12 droidu systemd[1]: rtl.service: Failed with result 'exit-code'.
Jun 19 21:02:42 droidu systemd[1]: rtl.service: Scheduled restart job, restart counter is at 1287.
Jun 19 21:02:42 droidu systemd[1]: Started rtl.service - Ride the Lightning.
Jun 19 21:02:43 droidu node[25706]: node:internal/modules/esm/resolve:215
Jun 19 21:02:43 droidu node[25706]:   const resolvedOption = FSLegacyMainResolve(packageJsonUrlString, packageConfig.main, baseStringified);
Jun 19 21:02:43 droidu node[25706]:                          ^
Jun 19 21:02:43 droidu node[25706]: Error: Cannot find package '/home/rtl/RTL/node_modules/request-promise/package.json' imported from /home/rtl/RTL/backend/utils/common.js
Jun 19 21:02:43 droidu node[25706]:     at legacyMainResolve (node:internal/modules/esm/resolve:215:26)
Jun 19 21:02:43 droidu node[25706]:     at packageResolve (node:internal/modules/esm/resolve:841:14)
Jun 19 21:02:43 droidu node[25706]:     at moduleResolve (node:internal/modules/esm/resolve:927:18)
Jun 19 21:02:43 droidu node[25706]:     at defaultResolve (node:internal/modules/esm/resolve:1157:11)
Jun 19 21:02:43 droidu node[25706]:     at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:383:12)
Jun 19 21:02:43 droidu node[25706]:     at ModuleLoader.resolve (node:internal/modules/esm/loader:352:25)
Jun 19 21:02:43 droidu node[25706]:     at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:227:38)
Jun 19 21:02:43 droidu node[25706]:     at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:87:39)
Jun 19 21:02:43 droidu node[25706]:     at link (node:internal/modules/esm/module_job:86:36) {
Jun 19 21:02:43 droidu node[25706]:   code: 'ERR_MODULE_NOT_FOUND'
Jun 19 21:02:43 droidu node[25706]: }
Jun 19 21:02:43 droidu node[25706]: Node.js v20.14.0
Jun 19 21:02:43 droidu systemd[1]: rtl.service: Main process exited, code=exited, status=1/FAILURE
Jun 19 21:02:43 droidu systemd[1]: rtl.service: Failed with result 'exit-code'.

**Reply by ShahanaFarooqui** [reply]
The error ERR_MODULE_NOT_FOUND is related to one of the required dependency `request-promise` for RTL. Did you run `npm install --omit=dev --legacy-peer-deps`?

**Reply by ghost** [reply]
I did.

**Reply by ShahanaFarooqui** [reply]
The error is definitely related to request-promise not found. So either the installation has corrupted somehow or for some reason your nodejs is unable to find it. 

- Try to explicitly install the library in RTL:
    - npm install request-promise
    - Restart your rtl service

- If it doesn't fix the issue, try with a fresh install:
    - Delete node_modules folder in the RTL
    - npm cache clean
    - `npm install --omit=dev --legacy-peer-deps`
    - Restart your rtl service

Do you manage multiple node versions? If yes, you might have to check your node version manager's settings.

**Reply by ghost** [reply]
Tried both and even restart the node but still same error related to request-promise. I am running only node 20.

**Reply by ghost** [reply]
Just tried removing node 20 and installing node 18, now different errors:
node:internal/errors:496
    ErrorCaptureStackTrace(err);
    ^

Error [ERR_INTERNAL_ASSERTION]: Code: ERR_MODULE_NOT_FOUND; The provided arguments length (2) does not match the required ones (3).
This is caused by either a bug in Node.js or incorrect usage of Node.js internals.
Please open an issue with this stack trace at https://github.com/nodejs/node/issues

    at new NodeError (node:internal/errors:405:5)
    at assert (node:internal/assert:14:11)
    at getMessage (node:internal/errors:451:5)
    at new NodeError (node:internal/errors:379:21)
    at legacyMainResolve (node:internal/modules/esm/resolve:218:9)
    at packageResolve (node:internal/modules/esm/resolve:903:14)
    at moduleResolve (node:internal/modules/esm/resolve:973:20)
    at defaultResolve (node:internal/modules/esm/resolve:1193:11)
    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:403:12)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:372:25) {
  code: 'ERR_INTERNAL_ASSERTION'
}

Node.js v18.19.1

**Reply by ghost** [reply]
What's odd is that I just installed btcrpc explorer and it works fine, should I remove rtl user and folders and start all over from scratch?

**Reply by ShahanaFarooqui** [reply]
I believe that the issue is related to nodejs version conflict. Removing RTL might not help here. 

Your working btcrpc explorer might be due to its compatibility with both conflicting versions.

Can you try to explicitly set your nodejs version before setting up RTL. Something like `nvm use 18.20.3`. 

Note that my default nodejs version is incompatible with RTL, so I am explicitly asking nodejs to use a different version than my default.

![Screenshot from 2024-06-20 13-12-18](https://github.com/Ride-The-Lightning/RTL/assets/23240721/a06e3f1b-421f-46cd-860b-942ccda54f40)

**Reply by ghost** [reply]
Command 'nvm' not found

**Reply by ghost** [reply]
just make it work..not sure why. But it did same thing as the problem on another node, this one is from a friend, on mine I basically did an upgrade as described here: https://raspibolt.org/guide/lightning/web-app.html#for-the-future-rtl-upgrade
it simply worked after it... before and after it was 0.15.1, the only explanation I have are my prayers to Satoshi

**Reply by ShahanaFarooqui** [reply]
Whatever works :D
